src/Engines/Nunjucks.js

```diff
@@ -4,6 +4,78 @@ const TemplatePath = require("../TemplatePath");
 const EleventyErrorUtil = require("../EleventyErrorUtil");
 const EleventyBaseError = require("../EleventyBaseError");
 
+/*
+ * HACKITYHACKHACKHACKHACK
+ */
+let inc = ((i) => {
+  return () => {
+    return i++;
+  };
+})(0);
+let pathMap = new Map();
+
+let clearCache = (reason = "") => {
+  if (pathMap.size > 0) {
+    pathMap.clear();
+  }
+};
+
+let clearAndWrap = (name, obj = NunjucksLib.Environment) => {
+  let orig = obj.prototype[name];
+  obj.prototype[name] = function (...args) {
+    clearCache(name);
+    return orig.call(this, ...args);
+  };
+};
+
+clearAndWrap("addExtension");
+clearAndWrap("removeExtension");
+
+(function () {
+  // IFFE to prevent temp var leakage
+  let getKey = (obj) => {
+    let eids = obj.env.extensionsList.reduce((v, c) => {
+      return c.__id ? v + " : " + c.__id : v;
+    }, "");
+    let k =
+      `${obj.path || obj.tmplStr} :: ${obj.tmplStr.length} :: ` +
+      `${obj.env.asyncFilters.length} :: (${eids})`; /* +
+            (
+              (obj.env.extensionsList.length) ? 
+                `${Object.keys(obj.env.extensions).join("|")} :: (${eids})` :
+                ""
+            );
+            */
+    return k;
+  };
+  let tc = NunjucksLib.Template.prototype._compile;
+  NunjucksLib.Template.prototype._compile = function (...args) {
```

Comment:
```suggestion
  NunjucksLib.Template.prototype._compile = function _compile(...args) {
```
Random drive-by feedback -- naming function might make call stacks easier to debug depending on your debugger!

Also, sorry if this is noise or wrong -- please let me know!

Reply:
A good point. Will address.

Reply:
Addressed in follow-up.